components/tools/ToolsList.js

```diff
@@ -1,12 +1,25 @@
 import ToolsCard from './ToolsCard';
 import Heading from '../typography/Heading'
+import { useRef, useEffect } from 'react';
 import Paragraph from '../typography/Paragraph'
 export default function toolsList({ toolsData }) {
+  const categoryRefs = useRef({});
+
+  useEffect(() => {
+    const hash = window.location.hash;
+    if (hash) {
+      const decodedHash = decodeURIComponent(hash.slice(1));
+      const categoryRef = categoryRefs.current[decodedHash];
+      if (categoryRef) {
+        categoryRef.scrollIntoView({ behavior: 'smooth', block: 'start' });
+      }
+    }
+  }, []);
```

Comment:
```suggestion
  }, []);
  
```

Reply:
What changes to add here ?

Reply:
just add a newline at the end of the file

Reply:
Done Sir